src/main/java/io/antmedia/rest/RestServiceBase.java

```diff
@@ -406,24 +407,50 @@ protected Result updateStreamSource(String streamId, Broadcast broadcast, String
 				}
 			}
 
+			result = getDataStore().updateBroadcastFields(streamId, broadcast);
+
+			if(result) {
+
+				Broadcast fetchedBroadcast = getDataStore().get(streamId);
+				getDataStore().removeAllEndpoints(fetchedBroadcast.getStreamId());
+
+				if (socialNetworksToPublish != null && socialNetworksToPublish.length() > 0) {
+					addSocialEndpoints(fetchedBroadcast, socialNetworksToPublish);
+				}
+
+				getApplication().startStreaming(fetchedBroadcast);
+
+			}
+		}
+		return new Result(result);
+	}
+
+	public boolean checkStopStreaming (String streamId, Broadcast broadcast)
+	{
+		// If broadcast status is broadcasting, this will force stop the streaming.
+		if(getDataStore().get(streamId).getStatus().equals(AntMediaApplicationAdapter.BROADCAST_STATUS_BROADCASTING)) {
+			return getApplication().stopStreaming(broadcast).isSuccess();
+		}
+		else
+		{
+			// If broadcast status is stopped, this will return true. 
+			return true;
+		}
+
+	}
+
+	public void waitStopStreaming(String streamId, Boolean resultStopStreaming) {
+
+		// Broadcast status finished is not enough to be sure about broadcast's status.
+		while (!getDataStore().get(streamId).getStatus().equals(AntMediaApplicationAdapter.BROADCAST_STATUS_FINISHED) && !resultStopStreaming.equals(true)) {
```

Comment:
It can stay in wait condition infinitely. Please write some logs and exit the loop if stream has not finished after some time like 5 seconds

Reply:
Yes, it makes sense.